src/main/java/net/smartcosmos/dao/metadata/impl/MetadataPersistenceService.java

```diff
@@ -130,6 +128,23 @@ private boolean alreadyExists(UUID tenantId, String ownerType, UUID ownerId, Map
         return Optional.empty();
     }
 
+    private List<MetadataEntity> getMetadataEntities(Map<String, Object> metadataMap, Set<String> keys, MetadataOwnerEntity owner) {
+        return keys.stream()
+                    .map(key -> {
+                        Object object = metadataMap.get(key);
+                        String value = MetadataValueParser.getValue(object);
+                        MetadataDataType dataType = MetadataValueParser.getDataType(object);
+
+                        return MetadataEntity.builder()
+                            .owner(owner)
+                            .keyName(key)
+                            .value(value)
+                            .dataType(dataType)
+                            .build();
+                    })
```

Comment:
Code style question: do we want to use anonymous internal functions inside lambdas or define a named private function instead for better readability?